crates/oxc_linter/src/rules/eslint/no_named_default.rs

```diff
@@ -0,0 +1,69 @@
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, module_record::ImportImportName, rule::Rule};
+
+fn no_named_default_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Replace default import with named import.")
+        .with_help("Forbid named default exports.")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoNamedDefault;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Reports use of a default export as a locally named import.
+    ///
+    /// ### Why is this bad?
+    /// Rationale: the syntax exists to import default exports expressively, let's use it.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// // message: Using exported name 'bar' as identifier for default export.
+    /// import { default as foo } from './foo.js';
+    /// import { default as foo, bar } from './foo.js';
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// import foo from './foo.js';
+    /// import foo, { bar } from './foo.js';
+    /// ```
+    NoNamedDefault,
+    style,
+    pending  // TODO: describe fix capabilities. Remove if no fix can be done,
+             // keep at 'pending' if you think one could be added but don't know how.
+             // Options are 'fix', 'fix_dangerous', 'suggestion', and 'conditional_fix_suggestion'
+);
+
+impl Rule for NoNamedDefault {
+    fn run_once(&self, ctx: &LintContext) {
+        ctx.module_record().import_entries.iter().for_each(|entry| {
+            let ImportImportName::Name(import_name) = &entry.import_name else {
+                return;
+            };
+            if import_name.name() == "default" {
+                ctx.diagnostic(no_named_default_diagnostic(entry.module_request.span()));
+            }
+        });
+    }
+}
+
+#[test]
+fn test() {
```

Comment:
for the reviewer, the original rule has differnt tests:

https://github.com/import-js/eslint-plugin-import/blob/main/tests/src/rules/no-named-default.js

Reply:
I edited the exemple to use the one in the original test. 
I just omit the typeof test. I don't know how to make it pass without decreasing performances

Reply:
can you include the typeof test and just comment it out with a reason? else it's confuring as to why it's commented out